lib/mkFenixToolchain.nix

```diff
@@ -30,6 +30,7 @@ in
 , clang-unwrapped ? defaultClangUnwrapped
 , stdenv ? defaultStdenv
 , isLintShell ? false
+, craneArgs ? { }
```

Comment:
Is this really needed? I thought we'd just modify `mergeArgs` to scan through `buildInputs` and `buildNativeInputs` and apply `pkgs` over functions there.

Reply:
I did try the `mergeArgs` route but it was a pain passing the `buildInputs` and `nativeBuildInputs` across different targets.
Resolving the packages as early as possible to pass it directly to `mkFenixToolchain` ended up working smoothly